core/internal/filetransfer/file_transfers.go

```diff
@@ -0,0 +1,38 @@
+package filetransfer
+
+import (
+	"github.com/hashicorp/go-retryablehttp"
+	"github.com/wandb/wandb/core/pkg/observability"
+)
+
+type FileTransfer interface {
+	Upload(task *Task) error
+	Download(task *Task) error
+}
+
+// FileTransfers is a collection of file transfers by upload destination type.
+type FileTransfers struct {
+	// Default makes an HTTP request to the destination URL with the file contents.
```

Comment:
specifically, it's a PUT request. I think it would be good to include that here!

Reply:
I didn't use PUT because for filetransfer because `download` uses a GET request instead of PUT? unless PUT is a more general term

Reply:
ah good point, I was too focused on uploads and forgot about downloads. Fine to leave as is!